pkg/platform/kube/functionres/lazy.go

```diff
@@ -998,6 +1033,74 @@ func (lc *lazyClient) createOrUpdateIngress(functionLabels labels.Set,
 	return resource.(*ext_v1beta1.Ingress), err
 }
 
+func (lc *lazyClient) deleteExistingCronJobs(functionName, functionNamespace string) error {
+	return lc.kubeClientSet.BatchV1beta1().
+		CronJobs(functionNamespace).
+		DeleteCollection(&meta_v1.DeleteOptions{},
+			meta_v1.ListOptions{LabelSelector: fmt.Sprintf("nuclio.io/function-name=%s", functionName)})
+}
+
+func (lc *lazyClient) createOrUpdateCronJob(functionLabels labels.Set,
+	function *nuclioio.NuclioFunction,
+	resources Resources,
+	triggerName string,
+	cronTrigger functionconfig.Trigger) (*v1beta1.CronJob, error) {
+
+	getCronJob := func() (interface{}, error) {
+		return lc.kubeClientSet.BatchV1beta1().
+			CronJobs(function.Namespace).
+			Get(kube.CronJobNameFromFunctionName(function.Name, triggerName), meta_v1.GetOptions{})
+	}
+
+	cronJobIsDeleting := func(resource interface{}) bool {
+		return (resource).(*v1beta1.CronJob).ObjectMeta.DeletionTimestamp != nil
+	}
+
+	createCronJob := func() (interface{}, error) {
+		cronJobMeta := meta_v1.ObjectMeta{
+			Name:      kube.CronJobNameFromFunctionName(function.Name, triggerName),
+			Namespace: function.Namespace,
+			Labels:    functionLabels,
+		}
+
+		cronJobSpec := v1beta1.CronJobSpec{}
+		if err := lc.populateCronJobConfig(functionLabels, function, resources, cronTrigger, &cronJobMeta, &cronJobSpec); err != nil {
+			return nil, errors.Wrap(err, "Failed to populate cron job spec")
+		}
+
+		resultCronJob, err := lc.kubeClientSet.BatchV1beta1().
+			CronJobs(function.Namespace).
+			Create(&v1beta1.CronJob{
+				ObjectMeta: cronJobMeta,
+				Spec:       cronJobSpec,
+			})
+
+		return resultCronJob, err
+	}
+
+	// should never perform update - because all cron jobs are deleted before creation of new ones
+	updateCronJob := func(resource interface{}) (interface{}, error) {
+		lc.logger.WarnWith("This log message should never be shown, means some cron job still exists when it should not")
```

Comment:
This code should never be ran - because before reaching here, all previous cron trigger cron jobs should be removed - so no update should be done
this is here for debugging reasons in case of a bug